docs/DeveloperGuide.md

```diff
@@ -25,14 +29,213 @@
+       Expected: Displays all visits for the specified patient.
+
+    2. (Positive) Test Case (by diagnosis): `findDiagnosis Fever` <br>
+       Expected: Lists all patients who were diagnosed with fever.
+
+    3. (Positive) Test Case (by medication): `findMedication Paracetamol` <br>
+       Expected: Lists all patients who were prescribed paracetamol.
+
+### Appointment Management
+
+1. Adding appointments
+    1. (Positive) Test Case: `appointment n/John Doe ic/S9876543A date/01-04-2025 time/12:00` <br>
+       Expected: Appointment is added if timeslot is available.
+
+    2. (Negative) Test Case (clash): Add appointment at same date/time as existing appointment
+       Expected: Error message suggesting next available timeslot.
+
+2. Listing appointments
+    1. (Positive) Test Case: `listAppointments` <br>
+       Expected: Shows all future appointments chronologically.
+
+3. Finding appointments
+    1. (Positive) Test Case: `findAppointment ic/S9876543A` <br>
+       Expected: Shows all appointments for the specified patient.
+
+4. Deleting appointments
+    1. (Positive) Test Case: `deleteAppointment ic/S9876543A date/01-04-2025 time/12:00` <br>
+       Expected: Specified appointment is deleted.
+
+    2. (Negative) Test Case (non-existent appointment): Delete appointment that doesn't exist <br>
+       Expected: Error message that appointment doesn't exist.
+
+### Data Persistence (Saving and Loading)
+
+1. Automatic Storage
+    1. Test case: Add/edit/delete records, then restart application
+        - Expected: All changes are preserved after restart
+        - Note: Data is automatically saved to `bookbob_data.txt` in the `data` folder (same directory as Bookbob.jar)
+
+2. File Management
+    1. Test case: Manually View saved data
+        - Action: Navigate to `data` folder (same directory as BookBob.jar) and open `bookbob_data.txt`
+        - Note: Do not manually modify the file contents to prevent data corruption. Saving and loading data is automated as long as file is not corrupted.
 
-## Instructions for manual testing
+    2. Test case: Delete data file and restart application
+        - Action: Delete `bookbob_data.txt` from the `data` folder
+        - Expected: New `bookbob_data.txt` file is automatically generated. The missing text file will not result in any error as files will be generated automatically.
+        - Note: This can be used to start with a fresh database if needed
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+3. Error Handling
+    1. Test case: Corrupt the data file manually, then start application
+        - Expected: Error message is shown about invalid data
+        - Note: BookBob will continue to function
+        - Recovery options:
+            * Option 1: Manually remove corrupted lines from `bookbob_data.txt`
+            * Option 2: Delete `bookbob_data.txt` to start afresh
```

Comment:
is it necessary to expand on recovery options when the data file is corrupted? Should we add specific steps or a short troubleshooting guide to assist users in recovering from data corruption errors?